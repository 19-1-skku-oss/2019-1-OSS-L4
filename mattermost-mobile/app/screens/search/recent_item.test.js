// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import RecentItem from './recent_item';

describe('Search RecentItem', () => {
    const item = {
        terms: 'test',
    };

    const baseProps = {
        item,
        removeSearchTerms: jest.fn(),
        setRecentValue: jest.fn(),
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot and respond to events', () => {
        const wrapper = shallow(
            <RecentItem {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
        wrapper.find('TouchableHighlight').first().props().onPress();
        expect(baseProps.setRecentValue).toHaveBeenCalledTimes(1);
        expect(baseProps.setRecentValue).toHaveBeenCalledWith(item);
        wrapper.find('TouchableOpacity').first().props().onPress();
        expect(baseProps.setRecentValue).toHaveBeenCalledTimes(1);
        expect(baseProps.setRecentValue).toHaveBeenCalledWith(item);
    });
});
