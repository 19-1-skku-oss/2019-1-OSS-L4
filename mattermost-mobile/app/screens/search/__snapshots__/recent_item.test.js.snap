// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search RecentItem should match snapshot and respond to events 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  onPress={[Function]}
  underlayColor="rgba(69,120,191,0.5)"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 42,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3d3c40",
          "flex": 1,
          "fontSize": 14,
          "height": 20,
          "paddingHorizontal": 16,
        }
      }
    >
      test
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 42,
          "justifyContent": "center",
          "width": 50,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="rgba(61,60,64,0.6)"
        name="ios-close-circle-outline"
        size={20}
      />
    </TouchableOpacity>
  </View>
</TouchableHighlight>
`;
